modules/002-deckhouse/templates/validation.yaml

```diff
@@ -99,4 +99,47 @@ spec:
     objectSelector:
       matchLabels:
         heritage: deckhouse
+---
+{{- if semverCompare ">= 1.30" .Values.global.discovery.kubernetesVersion }}
+apiVersion: admissionregistration.k8s.io/v1
+{{- else if semverCompare ">= 1.28" .Values.global.discovery.kubernetesVersion }}
+apiVersion: admissionregistration.k8s.io/v1beta1
+{{- else }}
+apiVersion: admissionregistration.k8s.io/v1alpha1
+{{- end }}
+{{- $policyName := "machines.deckhouse.io" }}
+kind: ValidatingAdmissionPolicy
+metadata:
+  name: {{ $policyName }}
+  {{- include "helm_lib_module_labels" (list . (dict "app" "deckhouse") ) | nindent 2 }}
+spec:
+  failurePolicy: Fail
+  matchConstraints:
+    resourceRules:
+      - apiGroups: ["machine.sapcloud.io", "cluster.x-k8s.io"]
+        apiVersions: ["*"]
+        operations: ["DELETE"]
+        resources: ["*"]
+  matchConditions:
+    - name: 'allow-only-for-d8-accounts'
+      expression: '!(request.userInfo.username.startsWith("system:serviceaccount:d8-") || request.userInfo.username == "kubernetes-admin")'
+  validations:
+    - expression: 'false'
+      reason: Forbidden
+      messageExpression: '''Operation on machine object is forbidden for this user by ValidatingAdmissionPolicy {{ $policyName }}'''
```

Comment:
what is the target milestone? As I remember, matchConditions option isn't supported in 1.26

Reply:
fixed